tests/subsys/rtio/rtio_api/src/test_rtio_spsc.c

```diff
@@ -229,6 +229,8 @@ ZTEST(rtio_spsc, test_spsc_throughput)
 
 	uint32_t *x, *y;
 
+	rtio_spsc_reset(&spsc);
+
```

Comment:
Can you instead create a `before` function for the test suite and do this there?

Reply:
Please correct me if I am wrong. Can the `before` function be applied for only the last two test cases (`test_spsc_threaded` and `test_spsc_throughput`) instead of for the whole test suite? `spsc` is only used by the last two test cases

Reply:
AFAIK, this might be a test case specific issue, not a generic resource alloc/free one for the whole test suite?